### DISCUSSION TITLE
Mapper implementation with Extension functions

### DISCUSSION THREAD

**Discussion by marcosscarpim** [igorescodro/alkaa#427] [Ideas]
What do you think about creating Mappers between modules using Extension functions for Model classes? I saw this implementation recently and it looked so clean in code.

I mean, instead of creating a "CategoryMapper" class with methods for conversion, we create extension function in this file:
''''
fun RepoCategory.toLocal(repoCategory: RepoCategory): LocalCategory = LocalCategory(...)
''''

**Comment by igorescodro** [comment]
Hello, Marcos!

In the past, we already have a discussion on how to better create mappers. The reason we decided to use a mapper `class` instead of a mapper `extension function` is due to architectural reasons. I agree that [models shouldn't be anemic](https://martinfowler.com/bliki/AnemicDomainModel.html), however, this kind of extension function implies that a model must know about the internals of a model in another layer. That's the reason we decided to extract a class in between that knows both representations.

With that said, I don't think there is a right or wrong solution here. It's a matter of preference because the boilerplate will need to be stored somewhere.

Thank you for your insight!
